packages/data-provider/src/controllers/submissionController.ts

```diff
@@ -212,6 +214,39 @@ const controller = (dependencies: BaseDependencies) => {
 				next(error);
 			}
 		},
+		deleteEntityName: async (
+			req: Request<{ submissionId: string; entityName: string }>,
+			res: Response,
+			next: NextFunction,
+		) => {
+			try {
+				const submissionId = Number(req.params.submissionId);
+				const entityName = req.params.entityName;
+
+				if (isNaN(submissionId)) {
+					throw new BadRequest('Invalid submissionId number format');
+				}
```

Comment:
doubtful there'll ever be a valid ID 0, so this should account for `req.params.submissionId` being null or '0' or empty strings too.

Also, error now shows consistent messaging with the other errors in this endpoint. e.g. "Request ..."

```suggestion
				if (isNaN(submissionId) || submissionId === 0) {
					throw new BadRequest('Request provided an invalid submission ID');
				}
```

Reply:
That's a good point thanks. Extracted this ID validation on a separate function